src/darwin/Framework/CHIP/MTRDevice.h

```diff
@@ -240,6 +240,90 @@ typedef NS_ENUM(NSUInteger, MTRDeviceState) {
                                       completion:(MTRDeviceOpenCommissioningWindowHandler)completion
     MTR_AVAILABLE(ios(17.0), macos(14.0), watchos(10.0), tvos(17.0));
 
+/**
+ *
+ * This set of functions allows clients to store metadata for either an entire device or for a specific endpoint.
+ *
+ * Notes:
+ *   • Client data will be removed automatically when devices are deleted from the fabric
+ *   • Supported client data object types are currently only:
+ *         NSData, NSString, NSArray, NSDictionary, NSNumber
+ */
+
+/**
+ *
+ * List of all client data types supported
+ *
+ */
+- (NSArray *)supportedClientDataClasses MTR_NEWLY_AVAILABLE;
+
+/**
+ *
+ * List of all client data keys stored
+ *
+ */
+- (NSArray * _Nullable)clientDataKeys MTR_NEWLY_AVAILABLE;
```

Comment:
Likewise, this should either document what the things in the NSArray are, or have it be `NSArray<Whatever>`, no?

Also, when does this return nil?